### DISCUSSION TITLE
My txt and MD file content changed when uplopded to minio server.

### DISCUSSION THREAD

**Discussion by chenxiaoleizi** [minio/minio#20194] [Q&A] [answered]
This is my local txt file content
![图片](https://github.com/user-attachments/assets/854094a6-cf99-4c01-ba55-f9a7a359e48b)
and this is server side file 
![图片](https://github.com/user-attachments/assets/db4b1efc-6230-4f83-a8ae-3f34679fcdfa)

Looks like content changed when uploded to the server side。But the doc file and pdf file is not changed.
This is my request headers
![图片](https://github.com/user-attachments/assets/b68bbc39-8beb-4b14-9f69-5cbe9a8cd07e)

I use axios to upload.

**Comment by marktheunissen** [comment]
MinIO does not modify file contents like that, the problem will be somewhere in your usage of the library (Axios). Can you use the MinIO SDK: https://github.com/minio/minio-js ?

If you can't use it for some reason, then read the source code to see if you can spot the difference between the way we do it in the SDK vs. the way you've implemented.

**Reply by chenxiaoleizi** [reply]
I need to use ajax to upload the file form the browser. I forget some detail about the upload. The browser get an url from the java server (use java API `getPresignedObjectUrl` to generate the url),  , then the broser use ajax PUT method to upload the file with the url from java.

(The first upload plan is : the browser use ajax upload to java server, then java serve upload the file to the minio,but if the file is large, the java server is not stable, so i change the plan to the up)

**Reply by chenxiaoleizi** [reply, marked as answer]
```javascript
// Do not use FormData to wrap the file.
const formData = new FormData();
formData.append("file",  file);
```
I found the problem.The problem is the `FormData` ,i wrap the file into the `FormData`  object, but the minio not extra the file from `FormData`, so put the file into the http body directily  will solve the problem. @marktheunissen  Thanks for your replay.
